新加入的raft共识节点初始化时panic

**Describe the bug**
新加入的raft共识节点初始化时panic

**To Reproduce**
1. 3节点raft集群正常运行
2. 添加第4个共识节点
3. 启动第4个节点（该节点为空节点）时panic

**Additional context**
版本号：[390ba0d](https://github.com/33cn/plugin/commit/390ba0d331a9fd42e76298b402dd6dbcef676346)

```
root@9436e68cf795:~# ./chain33
INFO[04-26|10:19:30] current dir:                             dir=/root
INFO[04-26|10:19:30] current dir:                             dir=/root
INFO[04-26|10:19:30] local-app:1.0.0 chain33:6.4.0-390ba0d3 localdb:1.0.0 statedb:1.0.0
INFO[04-26|10:19:30] loading queue
INFO[04-26|10:19:30] loading mempool module
INFO[04-26|10:19:30] loading execs module
INFO[04-26|10:19:30] mempool piple line start                 module=mempool.base
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.delegateNum not found module=types
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.blockInterval not found module=types
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.continueBlockNum not found module=types
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.blockNumToUpdateDelegate not found module=types
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.registTopNHeightLimit not found module=types
EROR[04-26|10:19:30] chain config config.consensus.sub.dpos.updateTopNHeightLimit not found module=types
INFO[04-26|10:19:30] loading blockchain module
INFO[04-26|10:19:30] init::LoadBlockStoreHeight::database may be crash module=blockchain err=ErrHeightNotExist
INFO[04-26|10:19:30] load block height error, may be init database module=blockchain height=-1
EROR[04-26|10:19:30] listSeqCB                                module=blockchain err=ErrNotFound
INFO[04-26|10:19:30] InitIndexAndBestView                     module=blockchain cost=4.651129ms
INFO[04-26|10:19:30] GetDbVersion len(version)==0             module=blockchain submodule=store
INFO[04-26|10:19:30] SetDbVersion                             module=blockchain submodule=store blcokchain db version=1
EROR[04-26|10:19:30] GetLastTempBlockHeight                   module=blockchain err=ErrNotFound
INFO[04-26|10:19:30] loading store module
INFO[04-26|10:19:30] FastDownLoadBlocks                       module=blockchain submodule=syn curHeight=-1 lastTempHight=-1
INFO[04-26|10:19:30] pollLastHeader quit                      module=mempool.base
INFO[04-26|10:19:30] FastDownLoadBlocks task sleep 1 second ! module=blockchain submodule=syn
INFO[04-26|10:19:30] Enter store mavl                         module=store
INFO[04-26|10:19:30] chain upgrade start                      module=blockchain
INFO[04-26|10:19:30] SetUpgradeMeta                           module=blockchain submodule=store meta="version:\"1.0.0\" "
INFO[04-26|10:19:30] storedb upgrade start                    module=blockchain
INFO[04-26|10:19:30] upgrade all dapp                         module=blockchain
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=relay
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=relay
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=retrieve
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=retrieve
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=ticket
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=ticket
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=none
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=none
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=echo
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=echo
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=guess
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=guess
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=jsvm
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=jsvm
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=oracle
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=oracle
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=pokerbull
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=pokerbull
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=storage
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=storage
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=valnode
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=valnode
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=manage
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=manage
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=accountmanager
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=accountmanager
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=lottery
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=lottery
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=privacy
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=privacy
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=token
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=token
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=autonomy
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=autonomy
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=game
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=game
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=multisig
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=multisig
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=trade
INFO[04-26|10:19:30] UpgradeLocalDBV2 upgrade start           module=execs.trade to_version=2
INFO[04-26|10:19:30] UpgradeLocalDBV2 upgrade done            module=execs.trade
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=paracross
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=paracross
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=unfreeze
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=unfreeze
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=coins
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=coins
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=hashlock
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=hashlock
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=blackwhite
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=blackwhite
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=collateralize
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=collateralize
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=evm
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=evm
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=norm
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=norm
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=cert
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=cert
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=dpos
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=dpos
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=exchange
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=exchange
INFO[04-26|10:19:30] begin upgrade plugin                     module=execs name=issuance
INFO[04-26|10:19:30] upgrade                                  module=execs.base dapp=issuance
INFO[04-26|10:19:30] upgrade plugin success                   module=execs
INFO[04-26|10:19:30] chain reduce start                       module=blockchain
INFO[04-26|10:19:30] loading consensus module
INFO[04-26|10:19:30] Start to create raft cluster             module=raft
INFO[04-26|10:19:30] Enter consensus raft                     module=raft
INFO[04-26|10:19:30] Enter consensus raft
INFO[04-26|10:19:30] replaying WAL of member 4                module=raft
INFO[04-26|10:19:30] Enter SetQueue method of raft consensus  module=raft
INFO[04-26|10:19:30] Enter SetQueueClient method of consensus
2020-04-26 10:19:30.650072 I | rpc.Register: type storage has no exported methods of suitable type
2020-04-26 10:19:30.650164 I | rpc.Register: type accountmanager has no exported methods of suitable type
2020-04-26 10:19:30.651309 I | rpc.Register: type relay has no exported methods of suitable type
INFO[04-26|10:19:30] loading wallet module
INFO[04-26|10:19:30] loading WAL at term 0 and index 0        module=raft
raft2020/04/26 10:19:30 INFO: 4 became follower at term 0
raft2020/04/26 10:19:30 INFO: newRaft 4 [peers: [], term: 0, commit: 0, applied: 0, lastindex: 0, lastterm: 0]
raft2020/04/26 10:19:30 INFO: 4 became follower at term 1
2020-04-26 10:19:30.707225 I | rafthttp: starting peer 1...
2020-04-26 10:19:30.707778 I | rafthttp: started HTTP pipelining with peer 1
2020-04-26 10:19:30.710842 I | rafthttp: started streaming with peer 1 (writer)
2020-04-26 10:19:30.713107 I | rafthttp: started streaming with peer 1 (writer)
2020-04-26 10:19:30.717542 I | rafthttp: started peer 1
2020-04-26 10:19:30.721294 I | rafthttp: added peer 1
INFO[04-26|10:19:30] initMingTicketTicker                     module=wallet.ticket Duration=2m0s
2020-04-26 10:19:30.721481 I | rafthttp: starting peer 2...
2020-04-26 10:19:30.721659 I | rafthttp: started HTTP pipelining with peer 2
INFO[04-26|10:19:30] Begin auto mining                        module=wallet.ticket
2020-04-26 10:19:30.725733 I | rafthttp: started peer 2
2020-04-26 10:19:30.725827 I | rafthttp: added peer 2
2020-04-26 10:19:30.725847 I | rafthttp: starting peer 3...
2020-04-26 10:19:30.725878 I | rafthttp: started HTTP pipelining with peer 3
2020-04-26 10:19:30.726785 I | rafthttp: started streaming with peer 1 (stream MsgApp v2 reader)
2020-04-26 10:19:30.726891 I | rafthttp: started streaming with peer 3 (stream Message reader)
2020-04-26 10:19:30.728159 I | rafthttp: started peer 3
2020-04-26 10:19:30.728318 I | rafthttp: added peer 3
2020-04-26 10:19:30.729028 I | rafthttp: started streaming with peer 1 (stream Message reader)
INFO[04-26|10:19:30] consensus is not ticket, exit mining     module=wallet.ticket
INFO[04-26|10:19:30] loading p2p module
2020-04-26 10:19:30.735640 I | rafthttp: started streaming with peer 2 (writer)
2020-04-26 10:19:30.735655 I | rafthttp: started streaming with peer 2 (writer)
2020-04-26 10:19:30.735661 I | rafthttp: started streaming with peer 2 (stream MsgApp v2 reader)
2020-04-26 10:19:30.736190 I | rafthttp: started streaming with peer 2 (stream Message reader)
2020-04-26 10:19:30.736542 I | rafthttp: started streaming with peer 3 (writer)
2020-04-26 10:19:30.736569 I | rafthttp: started streaming with peer 3 (writer)
2020-04-26 10:19:30.736578 I | rafthttp: started streaming with peer 3 (stream MsgApp v2 reader)
INFO[04-26|10:19:30] End auto mining                          module=wallet.ticket
2020-04-26 10:19:30.758173 I | rafthttp: peer 3 became active
2020-04-26 10:19:30.758863 I | rafthttp: established a TCP streaming connection with peer 3 (stream Message reader)
2020-04-26 10:19:30.759076 I | rafthttp: established a TCP streaming connection with peer 3 (stream MsgApp v2 reader)
2020-04-26 10:19:30.759341 I | rafthttp: peer 1 became active
2020-04-26 10:19:30.759388 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message reader)
2020-04-26 10:19:30.759470 I | rafthttp: peer 2 became active
2020-04-26 10:19:30.759508 I | rafthttp: established a TCP streaming connection with peer 2 (stream MsgApp v2 reader)
2020-04-26 10:19:30.759718 I | rafthttp: established a TCP streaming connection with peer 1 (stream MsgApp v2 reader)
INFO[04-26|10:19:30] p2p                                      module=p2p Channel=256 Version=10030 IsTest=true
2020-04-26 10:19:30.763339 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message reader)
INFO[04-26|10:19:30] p2p                                      module=p2p InnerBounds=300
2020-04-26 10:19:30.793769 I | rafthttp: established a TCP streaming connection with peer 1 (stream MsgApp v2 writer)
2020-04-26 10:19:30.795337 I | rafthttp: established a TCP streaming connection with peer 1 (stream Message writer)
2020-04-26 10:19:30.802799 I | rafthttp: established a TCP streaming connection with peer 2 (stream Message writer)
2020-04-26 10:19:30.807869 I | rafthttp: established a TCP streaming connection with peer 3 (stream Message writer)
2020-04-26 10:19:30.809803 I | rafthttp: established a TCP streaming connection with peer 3 (stream MsgApp v2 writer)
2020-04-26 10:19:30.811795 I | rafthttp: established a TCP streaming connection with peer 2 (stream MsgApp v2 writer)
raft2020/04/26 10:19:30 INFO: 4 [term: 1] received a MsgHeartbeat message with higher term from 2 [term: 2]
raft2020/04/26 10:19:30 INFO: 4 became follower at term 2
raft2020/04/26 10:19:30 INFO: raft.node: 4 elected leader 2 at term 2
INFO[04-26|10:19:30] newListener                              module=p2p localPort=13802
INFO[04-26|10:19:30] healthCheck start                        addr=127.0.0.1:8805 inter=5 times=6
INFO[04-26|10:19:30] ShowTaskCapcity                          module=p2p Capcity=1000
INFO[04-26|10:19:30] Metrics data is not enabled to emit      module="chain33 metrics"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x50 pc=0x11891e2]

goroutine 262 [running]:
github.com/33cn/plugin/plugin/consensus/raft.(*Client).CheckBlockInfo(0xc0002002a0, 0xc000799440, 0x60)
 /Users/hxz/workspace/golang/src/github.com/33cn/plugin/plugin/consensus/raft/block.go:307 +0x202
github.com/33cn/plugin/plugin/consensus/raft.(*raftNode).checkEntries(0xc0008de4e0, 0xc000509800, 0x48, 0x4c, 0x1)
 /Users/hxz/workspace/golang/src/github.com/33cn/plugin/plugin/consensus/raft/raft.go:307 +0x103
github.com/33cn/plugin/plugin/consensus/raft.(*raftNode).serveChannels(0xc0008de4e0)
 /Users/hxz/workspace/golang/src/github.com/33cn/plugin/plugin/consensus/raft/raft.go:268 +0x460
created by github.com/33cn/plugin/plugin/consensus/raft.(*raftNode).startRaft
 /Users/hxz/workspace/golang/src/github.com/33cn/plugin/plugin/consensus/raft/raft.go:175 +0x5bf
```

--- a/plugin/consensus/raft/block.go
+++ b/plugin/consensus/raft/block.go
@@ -35,7 +35,6 @@ type Client struct {
 	validatorC  <-chan bool
 	ctx         context.Context
 	cancel      context.CancelFunc
-	once        sync.Once
 	blockInfo   *BlockInfo
 	mtx         sync.Mutex
 }
@@ -98,6 +97,7 @@ func (client *Client) recoverFromSnapshot(snapshot []byte) error {
 func (client *Client) SetQueueClient(c queue.Client) {
 	rlog.Info("Enter SetQueue method of raft consensus")
 	client.InitClient(c, func() {
+		client.InitBlock()
 	})
 	go client.EventLoop()
 	go client.readCommits(client.commitC, client.errorC)
@@ -244,10 +244,6 @@ func (client *Client) pollingTask() {
 		case <-client.ctx.Done():
 			return
 		case value, ok := <-client.validatorC:
-			//各个节点Block只初始化一次
-			client.once.Do(func() {
-				client.InitBlock()
-			})
 			if ok && !value {
 				rlog.Debug("================I'm not the validator node=============")
 				leader := mux.Load().(bool)

